### PR TITLE
Clean up more unneeded changes

### PR DESCRIPTION
**Pull request by jeanp413**
This PR fixes https://github.com/gitpod-io/gitpod/issues/3342 and https://github.com/gitpod-io/gitpod/issues/7330

### CODE REVIEW

**Comment by jeanp413** [comment]
@akosyakov for some reason this changes were on `gp-code/main` branch, do you remember why?
reverting them seems to fix the issues linked above where reloading after uninstalling an extension will install the extension again

**Comment by akosyakov** [comment]
It is done on purporse till we don't have full workspace backup since nothing is preserved between workspace restarts you will loose all extensions otherwise. It would be good to add a warning comment that noone remove it for now.

**Comment by jeanp413** [comment]
mmm not following, why would I loose extensions? as far as I understand when a workspace is restarted extensions will install again from `gitpod.yml` and then sync the extensions from the server
In any case, if we need to persist the `.vscode-remote` then maybe we could move it inside the `/workspace` folder using `VSCODE_AGENT_FOLDER` env var or if we just need the extension folder use the `--extensions-dir` server argument

**Comment by akosyakov** [comment]
> mmm not following, why would I loose extensions? as far as I understand when a workspace is restarted extensions will install again from gitpod.yml and then sync the remaining extensions from the server

yes, it is the point, with this PR it does not happen

> In any case, if we need to persist the .vscode-remote then maybe we could move it inside the /workspace folder using VSCODE_AGENT_FOLDER env var or if we just need the extension folder use the --extensions-dir server argument

I think we probably can, we need to check with @csweichel whether it is better to wait till full workspace backup
if we do it we need to think about old workspaces, i.e. how we deploy without breaking anyone

**Comment by jeanp413** [comment]
> I think we probably can, we need to check with @csweichel whether it is better to wait till full workspace backup
if we do it we need to think about old workspaces, i.e. how we deploy without breaking anyone

Well, we already break everyone loosing workbench state when we do a new cluster deployment (e.g. us21 -> us23), which is particularly annoying for me :smile: , so I'll vote to go this route.

**Comment by akosyakov** [comment]
> Well, we already break everyone by loosing workbench state when we do a new cluster deployment (e.g. us21 -> us23), which is particularly annoying for me 😄 , so I'll vote to go this route.

I'm not sure it is related to this code. Could you elaborate? As far as I understand whenever we switch clusters then the same will happen regardless of this PR.

**Comment by jeanp413** [comment]
yeah, that's a different issue, I was just using it as an example.
Maybe you don't notice because you start a new workspace every time, but I always use the same workspace so I can continue working where I left (previusly opened files), but when I new cluster is deployed I loose all my workbench state because the URL change.

**Comment by akosyakov** [comment]
> Maybe you don't notice because you start a new workspace every time, but I always use the same workspace so I can continue working where I left (previously opened files, opened views etc), but when a new cluster is deployed I loose all my workbench state because the URL changes.

😬

**Comment by akosyakov** [comment]
Anyway to move with this forward. We should come up with some deployment strategy which does not break existing workspaces. If you can do it I don't mind helping to deliver it.

**Comment by jeanp413** [comment]
How about adding some check here and if .vscode-remote exists in /home/gitpod we copy it to /workspace
We could remove this change after a month I guess that would be enough for the migration right?
https://github.com/gitpod-io/openvscode-server/blob/01fe9797ab77322756e9ffe7605d21382044bfa8/src/vs/server/remoteExtensionHostAgent.ts#L37-L54

**Comment by jeanp413** [comment]
@akosyakov proposed to just move the `.vscode-remote` to the `/workspace` folder and keep the current behavior/bug for about a month or so (to avoid breaking users extensions) and once all workpaces have the `.vscode-remote` localtion updated then we can properly fix it.
@csweichel is that Ok with you as people gets reporting it and complete workspace backup is not yet implemented, is there and ETA for it?

**Comment by csweichel** [comment]
@csweichel is that Ok with you as people get reporting this issue ([gitpod-io/gitpod#7330](https://github.com/gitpod-io/gitpod/issues/7330)) and complete workspace backup is not yet implemented, is there and ETA for it?

That move makes sense to me. Full workspace backup currently has no ETA.

**Comment by akosyakov** [comment]
> Maybe you don't notice because you start a new workspace every time, but I always use the same workspace so I can continue working where I left (previously opened files, opened views etc), but when a new cluster is deployed I loose all my workbench state because the URL changes.

@jeanp413 Could you investigate please whether we can fix it somehow with user data sync? As far as I know some UI data is synced as well, maybe we need to change how machine id is computed?

**Comment by jeanp413** [comment]
> @jeanp413 Could you investigate please whether we can fix it somehow with user data sync? As far as I know some UI data is synced as well, maybe we need to change how machine id is computed?

Don't think user data sync fit this use case as it is shared across all workspaces, that issue is about workbench state specific to each workspace and each browser (e.g. you can open the same workspace in chrome and firefox and the workbench layout can be different)

**Comment by akosyakov** [comment]
> Don't think user data sync fit this use case as it is shared across all workspaces, that issue is about workbench state specific to each workspace and each browser (e.g. you can open the same workspace in chrome and firefox and the workbench layout can be different)

What do they sync in UI state then? I'm a bit puzzled.

**Comment by jeanp413** [comment]
> What do they sync in UI state then? I'm a bit puzzled. 

Also some workbench state but you could say it's more general:
- more recent used commands
- state about the getting started walktrough from extensions
- what views are pinned to the activity bar or the pannel

File: src/vs/gitpod/browser/workbench/workbench.ts [hunk 1]
```diff
@@ -946,6 +946,7 @@ async function doStart(): Promise<IDisposable> {
 		},
 		settingsSyncOptions: {
 			enabled: true,
+			extensionsSyncStateVersion: info.instanceId,
```
Comment (jeanp413) [inline src/vs/gitpod/browser/workbench/workbench.ts:949]: Found this setting which is used to reset the the sync state if the version changes, this should happen when a workspace is restarted, so I'm using `info.instanceId` which is different on workspace restart

**Comment by jeanp413** [comment]
I'll merge this to test `extensionsSyncStateVersion` in insiders

**Pull request merged**